domain/src/DomainFieldManager.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\domain;
+
+use Drupal\Core\Entity\EntityForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Generic base class for handling hidden field options.
+ *
+ * Since domain options are restricted for various forms (users, nodes, source)
+ * we have a base class for handling common use cases. The details of each
+ * implementation are generally handled by a subclass and invoked within a
+ * hook_form_alter().
+ */
+class DomainFieldManager {
```

Comment:
This can probably be the base class, and other classes can extend it. That would remove the need to call the $service dynamically, which is fugly code.